pkg/volume/fc/fc_util.go

```diff
@@ -265,13 +271,87 @@ func (util *FCUtil) AttachDisk(b fcDiskMounter) (string, error) {
 	return devicePath, err
 }
 
-func (util *FCUtil) DetachDisk(c fcDiskUnmounter, devName string) error {
+// DetachDisk removes scsi device file such as /dev/sdX from the node.
+func (util *FCUtil) DetachDisk(io ioHandler, devName string) error {
 	// Remove scsi device from the node.
 	if !strings.HasPrefix(devName, "/dev/") {
 		return fmt.Errorf("fc detach disk: invalid device name: %s", devName)
 	}
 	arr := strings.Split(devName, "/")
 	dev := arr[len(arr)-1]
-	removeFromScsiSubsystem(dev, c.io)
+	removeFromScsiSubsystem(dev, io)
+	return nil
+}
+
+// DetachBlockFCDisk detaches a volume from kubelet node, removes scsi device file
+// such as /dev/sdX from the node, and then removes loopback for the scsi device.
+func (util *FCUtil) DetachBlockFCDisk(io ioHandler, mapPath, devicePath string) error {
+	// Check if devicePath is valid
+	if len(devicePath) != 0 {
+		if pathExists, pathErr := volumeutil.PathExists(devicePath); pathErr != nil {
+			return fmt.Errorf("Error checking if path exists: %v", pathErr)
+		} else if !pathExists {
+			glog.Warningf("Warning: Unmap skipped because path does not exist: %v", mapPath)
+			return nil
+		}
+	} else {
+		// TODO: FC plugin can't obtain the devicePath from kubelet becuase devicePath
+		// in volume object isn't updated when volume is attached to kubelet node.
+		glog.Infof("fc: devicePath is empty. Try to retreive FC configuration from global map path: %v", mapPath)
+	}
+	// Check if global map path is valid
+	// global map path examples:
+	//   wwn+lun: plugins/kubernetes.io/fc/volumeDevices/50060e801049cfd1-lun-0/
+	//   wwid: plugins/kubernetes.io/fc/volumeDevices/3600508b400105e210000900000490000/
+	if pathExists, pathErr := volumeutil.PathExists(mapPath); pathErr != nil {
```

Comment:
consolidate into a function to validate paths

Reply:
done.